docker-storage-setup.sh

```diff
@@ -136,12 +136,14 @@ should_enable_deferred_deletion() {
 platform_supports_deferred_deletion() {
         local deferred_deletion_supported=1
         trap cleanup_pipes EXIT
-        mkfifo $PIPE1
-        mkfifo $PIPE2
-        mount -o bind $TEMPDIR $TEMPDIR
+
         if [ ! -x "/usr/lib/docker-storage-setup/dss-child-read-write" ];then
            return 1
         fi
+
+        mkfifo $PIPE1
+        mkfifo $PIPE2
+        mount -o bind $TEMPDIR $TEMPDIR
         unshare -m /usr/lib/docker-storage-setup/dss-child-read-write $PIPE1 $PIPE2 &
```

Comment:
We could rely on the kernel to umount if we did the mount inside the unshare, no?